src/sqlfluff/rules/L022.py

```diff
@@ -194,12 +194,12 @@ def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
                             offset = 1
                             while line_idx - offset in comment_lines:
                                 offset += 1
+                            # If the offset - 1 equals the line_idx then there aren't
+                            # really any comment-only lines (ref #2945).
+                            # Reset to line_idx
                             fix_point = forward_slice[
-                                line_starts[line_idx - (offset - 1)]
+                                line_starts[line_idx - (offset - 1) or line_idx]
                             ]
-                        # Note: There is an edge case where this isn't enough, if
-                        # comments are in strange places, but we'll catch them on
-                        # the next iteration.
```

Comment:
Not sure if this is the only edge case, but didn't like this comment. A bit vague, and kind just what the rules did anyway. So removed it.